File: 53.maximum-subarray.cpp
/*
 * @lc app=leetcode id=53 lang=cpp
 *
 * [53] Maximum Subarray
 */

// @lc code=start
#include <bits/stdc++.h>
class Solution {
public:
    int maxSubArray(vector<int>& nums) {

        int sum=nums[0],maxSum=nums[0],p;
        for (int i = 1; i < nums.size(); i++)
        {
            p = sum+nums[i];
            sum = p>nums[i]?p:nums[i];
            if(sum>maxSum){
                maxSum = sum;
            }
        }
        return maxSum;
        
    }
};
// @lc code=end

File: 53.maximum-subarray.java
/*
 * @lc app=leetcode id=53 lang=java
 *
 * [53] Maximum Subarray
 */

// @lc code=start
class Solution {
    public int maxSubArray(int[] nums) {

       int   sum=nums[0]; 
       int maxSum = sum;
       int p;
       for (int i = 1; i < nums.length; i++) {
           
        
        p= nums[i]+sum;
        sum = p>nums[i]?p:nums[i];
        maxSum = Math.max(maxSum, sum);

       }
       return maxSum;

    }
}
// @lc code=end

